frontend/src/components/employees/index.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import Box from '@material-ui/core/Box';
+import Button from '@material-ui/core/Button';
+import Table from '@material-ui/core/Table';
+import TableHead from '@material-ui/core/TableHead';
+import TableBody from '@material-ui/core/TableBody';
+import TableRow from '@material-ui/core/TableRow';
+import TableCell from '@material-ui/core/TableCell';
+import toastr from 'toastr';
+import { StaffData, fetchStaff, selectStaff } from '../../features/staffSlice';
+import 'toastr/build/toastr.min.css';
+import employeesStyle from './employees.module.scss';
+
+const STAFF_FETCH_URL = 'http://localhost:8080/staff/get';
+const AUTHENTICATION_TOKEN = 'xx';
+const FULLTIME_HOURS = 37.5;
+
+interface TableColumn {
+  id: string;
+  name: string;
+  content: (o: StaffData) => any;
+}
+
+/**
+ * Functions which are called when add/edit/delete buttons are pressed.
+ * Note that the signature of addUser is different from editUser/deleteUser.
+ */
+const addUser = () => toastr.info('Add user');
+const editUser = (id: number) => () => toastr.info(`User ${id} edited`);
+const deleteUser = (id: number) => () => toastr.info(`User ${id} deleted`);
+
+/**
+ * Functions which extract the data to be displayed in the table columns
+ * from StaffData.
+ */
+const getName = ({ firstName, surname }: StaffData) =>
+  `${firstName} ${surname}`;
+const getJobTitle = ({ jobTitle }: StaffData) => jobTitle;
+const partFullTime = (fullTimeHours: number) => ({
+  contractedHours,
+}: StaffData) => (contractedHours >= fullTimeHours ? 'full' : 'part');
+const editUserButton = (editFunction: (o: number) => () => any) => ({
+  id,
+}: StaffData) => (
+  <button
+    type="button"
+    className={employeesStyle.editButton}
+    onClick={editFunction(id)}>
+    &#x1F589;
+  </button>
+);
+const removeUserButton = (removeFunction: (o: number) => () => any) => ({
```

Comment:
What would be your thoughts on having an extra line of space between functions? When functions span several lines it feels easier to read with extra space, though it may look unneccessary when functions are only 1 line.